src/duplicacy_snapshotmanager.go

```diff
@@ -2215,9 +2217,11 @@ func (manager *SnapshotManager) DownloadFile(path string, derivationKey string)
         return nil
     }
 
-    err = manager.snapshotCache.UploadFile(0, path, manager.fileChunk.GetBytes())
-    if err != nil {
-        LOG_WARN("DOWNLOAD_FILE_CACHE", "Failed to add the file %s to the snapshot cache: %v", path, err)
+    if !manager.config.dryRun {
```

Comment:
It might be ok to save the downloaded file in the snapshot cache in dry-run mode.